tss-esapi/src/structures/lists/command_code.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    constants::CommandCode,
+    tss2_esys::{TPM2_MAX_CAP_CC, TPML_CC},
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::{convert::TryFrom, ops::Deref};
+
+/// A list of command codes.
+#[derive(Debug, Clone, Default)]
+pub struct CommandCodeList {
+    command_codes: Vec<CommandCode>,
+}
+
+impl CommandCodeList {
+    pub const MAX_SIZE: usize = TPM2_MAX_CAP_CC as usize;
+    /// Creates a new CommandCodeList
+    pub const fn new() -> Self {
+        CommandCodeList {
+            command_codes: Vec::new(),
+        }
+    }
+
+    /// Adds a command code to the command code list.
+    pub fn add(&mut self, command_code: CommandCode) -> Result<()> {
+        if self.command_codes.len() + 1 > CommandCodeList::MAX_SIZE {
+            error!(
+                "Adding command code to list will make the list exceeded its maximum count(> {})",
+                CommandCodeList::MAX_SIZE
+            );
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        self.command_codes.push(command_code);
+        Ok(())
+    }
+
+    /// Returns the inner type.
+    pub fn into_inner(self) -> Vec<CommandCode> {
+        self.command_codes
+    }
```

Comment:
Every now and then I wish we had something like this implemented for every type.